spark/src/test/scala/org/apache/comet/CometCastSuite.scala

```diff
@@ -1030,7 +1030,9 @@ class CometCastSuite extends CometTestBase with AdaptiveSparkPlanHelper {
 
   test("cast between decimals with higher precision than source") {
     // cast between Decimal(10, 2) to Decimal(10,4)
-    castTest(generateDecimalsPrecision10Scale2(), DataTypes.createDecimalType(10, 4))
+    castTest(
```

Comment:
@manuzhang I pulled these test changes locally using latest from main branch and they pass for me, so I'm not sure that these tests reproduce the problem that the fix is intended to address?

Reply:
@andygrove I've reworked my PR. Could you please check again?